pyseas/maps/bivariate.py

```diff
@@ -0,0 +1,239 @@
+
+import numpy as np
+from matplotlib.colors import Normalize, LogNorm
+from matplotlib import cm
+import matplotlib.pyplot as plt
+import warnings
+from . import core
+from .. import props
+
+
+class BivariateColormap:
+
+    log_x = None
+    log_y = None
+
+    def __call__(self, X, Y, alpha=None):
+        """Return color for the pair of values X and Y.
+
+        Parameters
+        ----------
+        X, Y : float or np.array of float
+            Each pair of X, Y values is mapped to a color value.
+            The data should be in the range [0.0, 1.0].
+        alpha : float or np.array or float
+            A scalar in the range [0.0, 1.0] that specifies the 
+            opacity at each X, Y pair. If alpha is an array, it 
+            must match the shape of X and Y.   
+
+        Return
+        ------
+        np.array
+            The shape will be np.shape(X) + (4,) to accommodate RGBA
+            information.       
+        """
+
+
+class TransparencyBivariateColormap(BivariateColormap):
+    """Bivariate colormap combining standard colormap with transparency
+
+    The first value (`X`) is looked up in `base_cmap` and the result is
+    multiplied by the second value ('Y').
+
+    Parameters
+    ----------
+    base_cmap : matplotlib Colormap
+    log_x, log_y : bool, optional
+    """
+
+    def __init__(self, base_cmap, log_x=None, log_y=None):
+        self.base_cmap = base_cmap
+        self.log_x = log_x
+        self.log_y = log_y
+
+    # TODO: should make this parameterizable
+    def __call__(self, X, Y, alpha=None):
+        colors = self.base_cmap(X)
+        colors[..., 3] *= Y
+        if alpha is not None:
+            colors[..., 3] *= alpha
+        return colors
+
+
+class SummedBivariateColormap(BivariateColormap):
+    """Bivariate colormap summing two standard colormaps
+
+    The colors for `X` and `Y` are looked up in `cmap1` and `cmap2`
+    respectively and then their values are summed and clipped to the
+    range [0, 1]. 
+
+    N.B., the two colormaps should be chosen to be compatible with each
+    other, e.g. Reds and Blues or this will not work well.
+
+    Parameters
+    ----------
+    cmap1, cmap2 : matplotlib colormap
+    """
+
+    def __init__(self, cmap1, cmap2):
+        self.cmap1 = cmap1
+        self.cmap2 = cmap2
+
+    # TODO: should make this parameterizable
```

Comment:
Same here, does this need to be done before merging?

Reply:
Ditto.